pyomo/core/expr/visitor.py

```diff
@@ -1258,6 +1258,9 @@ def evaluate_expression(exp, exception=True, constant=False):
     finally:
         if clear_active:
            evaluate_expression.visitor_active = False
+    if ans.__class__ not in native_types and ans.is_numeric_type() is True:
```

Comment:
@jsiirola, the failing tests are because of this line. But I'm not sure what to do about it?